### PR TITLE
feat: ✨ kyber-pmm: inventory with swapped

### PR DESCRIPTION
**Pull request by thanhpp**
## Why did we need it?

- kyber-pmm needs to shift orderbooks of every related pool 

## Release Note

- InventoryWithSwapped
- Update kyber-pmm calc function

### CODE REVIEW

File: pkg/source/kyber-pmm/pool_simulator.go [hunk 1]
```diff
@@ -332,12 +350,12 @@ func getNewPriceLevelsState(
...
+// Inventory is an alias for swaplimit.InventoryWithSwapped
+// Deprecated: directly use swaplimit.InventoryWithSwapped.
+type Inventory = swaplimit.InventoryWithSwapped
 
 // NewInventory has key as token address and value as its balance.
-// Deprecated: directly use swaplimit.NewInventory.
+// Deprecated: directly use swaplimit.NewInventoryWithSwapped.
 func NewInventory(balance map[string]*big.Int) pool.SwapLimit {
-	return swaplimit.NewInventory(DexTypeKyberPMM, balance)
+	return swaplimit.NewInventoryWithSwapped(DexTypeKyberPMM, balance)
```
Comment (NgoKimPhu) [inline pkg/source/kyber-pmm/pool_simulator.go:360]: can remove now and directly use `swaplimit`

File: pkg/swaplimit/inventory_with_swapped.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"maps"
+	"math/big"
+	"sync"
+
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/source/pool"
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/util/bignumber"
+)
+
+type InventoryWithSwapped struct {
```
Comment (NgoKimPhu) [inline pkg/swaplimit/inventory_with_swapped.go:12]: `SwappedInventory`?

File: pkg/swaplimit/inventory_with_swapped.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	increasedTokenBalance, ok := k.balance[increaseTokenAddress]
+	if !ok {
+		increasedTokenBalance = new(big.Int).Set(increaseDelta)
+	} else {
+		increasedTokenBalance = new(big.Int).Add(increasedTokenBalance, increaseDelta)
+	}
+	k.balance[increaseTokenAddress] = increasedTokenBalance
+
+	return decreasedTokenBalance, increasedTokenBalance, nil
```
Comment (NgoKimPhu) [inline pkg/swaplimit/inventory_with_swapped.go:72]: refactor: extract `Inventory.updateLimit` from `Inventory.UpdateLimit` without lock and then call `k.Inventory.updateLimit` here

Comment (thanhpp) [reply pkg/swaplimit/inventory_with_swapped.go]: updated in https://github.com/KyberNetwork/kyberswap-dex-lib/pull/598/commits/94c367c7c84527ee30047d20e4059bf3f1f7b0ca

File: pkg/swaplimit/inventory_with_swapped.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+	return maps.Clone(k.swapped)
+}
+
+func (k *InventoryWithSwapped) UpdateLimit(decreaseTokenAddress, increaseTokenAddress string,
+	decreaseDelta, increaseDelta *big.Int) (*big.Int, *big.Int, error) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	swapped1, ok := k.swapped[decreaseTokenAddress]
```
[the remark this answers is not in the document]
Comment (thanhpp) [reply pkg/swaplimit/inventory_with_swapped.go]: updated in https://github.com/KyberNetwork/kyberswap-dex-lib/pull/598/commits/94c367c7c84527ee30047d20e4059bf3f1f7b0ca

File: pkg/swaplimit/inventory_with_swapped.go (hunk 2 above)
Comment (thanhpp) [reply pkg/swaplimit/inventory_with_swapped.go]: updated in https://github.com/KyberNetwork/kyberswap-dex-lib/pull/598/commits/94c367c7c84527ee30047d20e4059bf3f1f7b0ca

File: pkg/source/kyber-pmm/pool_simulator.go (hunk 1 above)
Comment (thanhpp) [reply pkg/source/kyber-pmm/pool_simulator.go]: updated in https://github.com/KyberNetwork/kyberswap-dex-lib/pull/598/commits/94c367c7c84527ee30047d20e4059bf3f1f7b0ca

File: pkg/source/kyber-pmm/pool_simulator.go [hunk 5]
```diff
@@ -145,33 +163,32 @@ func (p *PoolSimulator) UpdateBalance(params pool.UpdateBalanceParams) {
...
 
 	if swapDirection == SwapDirectionBaseToQuote {
-		amountInAfterDecimals := new(big.Float).Quo(
-			new(big.Float).SetInt(params.TokenAmountIn.Amount),
-			bignumber.TenPowDecimals(p.baseToken.Decimals),
-		)
+		// amountInAfterDecimals := new(big.Float).Quo(
+		// 	new(big.Float).SetInt(params.TokenAmountIn.Amount),
+		// 	bignumber.TenPowDecimals(p.baseToken.Decimals),
+		// )
```
Comment (NgoKimPhu) [inline pkg/source/kyber-pmm/pool_simulator.go:169]: remove all commented codes plz

**Review by NgoKimPhu — APPROVED** [review]

**Review by NgoKimPhu — APPROVED** [review]

**Pull request merged**